### PR TITLE
refactor: Physmon in pytest

### PR DESCRIPTION
**Pull request by paulgessinger**
This PR proposes reimplementing the physmon workflows inside pytest, where tests are defined to run the workflows.
This also allows me to define most of the extra steps, like file output comparisons and file existence checks can be orchestrated via python functions. This also allows me to provide a unified way to provide a local reference directory like

```
pytest -k "physmon" -v --physmon-reference-path local_references
```

and also to automatically update the references like

```
pytest -k "physmon" -v --physmon-update-references
```

The two can also be combined, which is nice!

### CODE REVIEW

**Comment by paulgessinger** [comment]
Draft for now, to gather feedback before potentially adding the other workflows.

What do you think @andiwand @benjaminhuth ?

File: Examples/Python/setup.sh.in
Comment (paulgessinger) [inline Examples/Python/setup.sh.in:1]: Unrelated, can factor out.

**Comment by andiwand** [comment]
I like it! I wonder if this should ultimately also go into `Examples/Python/tests`? we could either separate it by folders or mark the tests differently. at the end of the day all of these are higher level integration tests I feel like

**Comment by paulgessinger** [comment]
@andiwand Ultimately, I don't think it matters from a technical standpoint, only from an organizational one.

**Comment by benjaminhuth** [comment]
So I like it as well! Just one question: If I run `pytest` from the top level directory, does it execute all tests including physmon tests automatically?

**Comment by paulgessinger** [comment]
I'll need to revisit this another time.

**Pull request closed without merging**